apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/V1/PgNumericTests.cs

```diff
@@ -0,0 +1,453 @@
+            }
+
+            PgNumeric numeric = PgNumeric.Parse(text);
+            Assert.Equal(text, numeric.ToString());
+
+            // Check that TryParse works too
+            Assert.True(PgNumeric.TryParse(text, out var numeric2));
+            Assert.Equal(numeric, numeric2);
+        }
+
+        [Theory]
+        [MemberData(nameof(InvalidStringInputData))]
+        public void Parse_InValid(string input)
+        {
+            Assert.Throws<FormatException>(() => PgNumeric.Parse(input));
+        }
+
+        [Theory]
+        [MemberData(nameof(ValidStringInputData))]
+        public void TryParse_Valid(string input)
+        {
+            var success = PgNumeric.TryParse(input, out PgNumeric number);
+            Assert.True(success);
+        }
+
+        [Theory]
+        [MemberData(nameof(InvalidStringInputData))]
+        public void TryParse_InValid(string input)
+        {
+            bool success = PgNumeric.TryParse(input, out PgNumeric number);
+            Assert.Equal(PgNumeric.Zero, number);
+            Assert.False(success);
+        }
+
+        [Theory]
+        [InlineData("NaN", "NaN")]
+        [InlineData("1.00", "1.00")]
+        [InlineData("1.000000", "1.000000")]
+        [InlineData("99999.999999000", "99999.999999000")]
+        [InlineData("-0.000001", "-0.000001")]
+        [InlineData("100", "100")]
+        [InlineData("1.33333333333333333", "1.33333333333333333")]
+        public void To_String(string input, string expected)
+        {
+            PgNumeric pgNumeric = PgNumeric.Parse(input);
+            Assert.Equal(expected, pgNumeric.ToString());
+        }
+
+        [Theory]
+        [MemberData(nameof(FromDecimalData))]
+        public void FromDecimal(decimal input, string expectedValue)
+        {
+            var numeric = PgNumeric.FromDecimal(input);
+            Assert.Equal(expectedValue, numeric.ToString());
+        }
+
+        [Theory]
+        [InlineData("0.00", "0")]
+        [InlineData("1.00", "1.00")]
+        [InlineData("1.01", "1.01")]
+        [InlineData("12345.123456000", "12345.123456000")]
+        [InlineData("-0.000001", "-0.000001")]
+        [InlineData("1234567891.123456789123", "1234567891.123456789123")]
+        [InlineData("-12345678912.12345678912", "-12345678912.12345678912")]
+        [InlineData("123456789123456789.12345678912", "123456789123456789.12345678912")]
+        [InlineData("12345678912345678912345678.1234567891234567891234", "12345678912345678912345678.1234567891234567891234")]
+        [InlineData("9999999999999999999999999999.999999999999999999999999", "9999999999999999999999999999.999999999999999999999999")]
+        [InlineData("100", "100")]
+        [InlineData("1.33333333333333333", "1.33333333333333333")]
+        public void ToDecimal(string input, string expectedValue)
+        {
+            var numeric = PgNumeric.Parse(input);
+            var decimalValue = numeric.ToDecimal(LossOfPrecisionHandling.Truncate);
```

Comment:
Yes. Now they do.

Reply:
Hi Jon, Please see if the updated test makes better sense of if any further change is desired.

Reply:
I'm still not seeing anything that tests actual truncation in an obvious way - that 9999999999999999999999999999.999999999999999999999999 ends up with truncated digits.

I can sort of see how the test is doing it, due to using decimal.Parse, but I think it would be clearer if the test case included the expected result of `decimal.ToString` - so we could really see what the result is.

I'm also confused as to how the test worked before, in the outdated version. That's what really worried me. Unfortunately I can't see the old version now, so I can't see *exactly* what was going on. Basically, I don't know whether a production code change is required or not.

(I'd also prefer the method names to give more information about what's going on - for example, "ToDecimal_LossOfPrecision" doesn't tell us what value of LossOfPrecision is being supplied, rather than just that it *is* being supplied.)

Reply:
Please see if these new test cases and test data makes sense. 
My understanding about tests working earlier is that when we parse 999[...].999[...] as decimal, since it is more than 29 digits, it rounds up to 10[...] and that is what was put as expected value as well. 
I have also changed the name of methods by appending _Truncate and _Throw to the test methods. Thanks